aten/src/ATen/native/quantized/cpu/qpool.cpp

```diff
@@ -395,9 +400,37 @@ Tensor quantized_max_pool2d(
   return qy;
 }
 
+// Quantized max_pool1d is a special case of the max_pool2d, with one of the
+// dimensions and kernels removed.
+Tensor quantized_max_pool1d(
+    const Tensor& qx,
+    IntArrayRef kernel_size,
+    IntArrayRef stride,
+    IntArrayRef padding,
+    IntArrayRef dilation,
+    bool ceil_mode) {
+  // (C, L) -> (C, 1, L) => kSqueezeDim = 1
+  // (N, C, L) -> (N, C, 1, L) => kSqueezeDim = 2
```

Comment:
just double checking, this works as expected across different memory formats, right?

Reply:
Can you elaborate -- are you asking if squeeze/unsqueeze works properly across the memory formats, or if this trick does? For the latter -- it does, that what they do for the FP pooling ops. As for the squeeze/unsqueeze -- not sure, need to check the ref

Reply:
yes, was just asking if this function will work as expected for various memory formats.  If it does - sg.